clusterctl/validation/validate_cluster_api_objects_test.go

```diff
@@ -0,0 +1,411 @@
+)
+
+var client *clientset.Clientset
+
+func TestMain(m *testing.M) {
+	testenv := test.NewTestEnvironment()
+	config := testenv.Start(apis.GetAllApiBuilders(), openapi.GetOpenAPIDefinitions)
+	client = clientset.NewForConfigOrDie(config)
+
+	code := m.Run()
+
+	testenv.Stop()
+	os.Exit(code)
+}
+
+func getClusterWithError(errorReason common.ClusterStatusError, errorMessage string) v1alpha1.Cluster {
+	return v1alpha1.Cluster{
+		Spec: v1alpha1.ClusterSpec{
+			ClusterNetwork: v1alpha1.ClusterNetworkingConfig{
+				Services: v1alpha1.NetworkRanges{
+					CIDRBlocks: []string{"10.96.0.0/12"},
+				},
+				Pods: v1alpha1.NetworkRanges{
+					CIDRBlocks: []string{"192.168.0.0/16"},
+				},
+				ServiceDomain: "cluster.local",
+			},
+		},
+		Status: v1alpha1.ClusterStatus{
+			ErrorReason:  errorReason,
+			ErrorMessage: errorMessage,
+		},
+	}
+}
+
+func getMachineWithError(nodeRef *v1.ObjectReference, errorReason *common.MachineStatusError, errorMessage *string) v1alpha1.Machine {
+	return v1alpha1.Machine{
+		Status: v1alpha1.MachineStatus{
+			NodeRef:      nodeRef,
+			ErrorReason:  errorReason,
+			ErrorMessage: errorMessage,
+		},
+	}
+}
+
+func TestGetClusterObjectWithNoCluster(t *testing.T) {
+	t.Run("Get cluster", func(t *testing.T) {
+		_, err := getClusterObject(client, "test-cluster", "get-cluster-object-with-no-cluster")
+		if err == nil {
+			t.Fatalf("Expect to get error, but got no returned error.")
+		}
+	})
+}
+
+func TestGetClusterObjectWithOneCluster(t *testing.T) {
+	testClusterName := "test-cluster"
+	testNamespace := "get-cluster-object-with-one-cluster"
+	clusterClient := client.ClusterV1alpha1().Clusters(testNamespace)
+	cluster := testutil.GetVanillaCluster()
+	cluster.Name = testClusterName
+	actualCluster, err := clusterClient.Create(&cluster)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer clusterClient.Delete(actualCluster.Name, &meta_v1.DeleteOptions{})
+	var testcases = []struct {
+		name        string
+		clusterName string
+		namespace   string
+		expectErr   bool
+	}{
+		{
+			name:        "Get cluster with name",
+			clusterName: testClusterName,
+			namespace:   testNamespace,
+			expectErr:   false,
+		},
+		{
+			name:        "Get cluster without name",
+			clusterName: "",
+			namespace:   testNamespace,
+			expectErr:   false,
+		},
+		{
+			name:        "Get cluster with name in different namespace",
+			clusterName: testClusterName,
+			namespace:   "different-namespace",
+			expectErr:   true,
+		},
+		{
+			name:        "Get cluster without name in different namespace",
+			clusterName: "",
+			namespace:   "different-namespace",
+			expectErr:   true,
+		},
+	}
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			cluster, err := getClusterObject(client, testcase.clusterName, testcase.namespace)
+			if (testcase.expectErr && err == nil) || (!testcase.expectErr && err != nil) {
+				t.Fatalf("Unexpected returned error. Got: %v, Want Err: %v", err, testcase.expectErr)
```

Comment:
Since expectErr is a boolean, this will read:
"Unexpected returned error. Got: error message, Want Err: true"
Is that the intended behavior? It seems like it would be more useful to print the error message that you expected